src/core/lombok/core/AnnotationProcessor.java

```diff
@@ -100,6 +97,40 @@ private static ProcessingEnvironment tryRecursivelyObtainJavacProcessingEnvironm
 		return null;
 	}
 	
+	/**
+	 * Gradle incremental processing
+	 */
+	private static Object tryGetDelegateField(Class<?> delegateClass, Object instance) {
+		try {
+			return Permit.getField(delegateClass, "delegate").get(instance);
+		} catch (Exception e) {
+			return null;
+		}
+	}
+	
+	/**
+	 * Kotlin incremental processing
+	 */
+	private static Object tryGetProcessingEnvField(Class<?> delegateClass, Object instance) {
+		try {
+			return Permit.getField(delegateClass, "processingEnv").get(instance);
+		} catch (Exception e) {
+			return null;
+		}
+	}
+	
+	/**
+	 * InteliJ >= 2020.3
+	 */
+	private static Object tryGetProxyDelegateToField(Class<?> delegateClass, Object instance) {
+		try {
+			InvocationHandler handler = Proxy.getInvocationHandler(instance);
+			return Permit.getField(handler.getClass(), "val$delegateTo").get(handler);
```

Comment:
I would suggest making this heuristic a bit more reliable and not depending on particular field name. Also all Proxy-related suff will remain hidden. 
Consider using the _org.jetbrains.jps.javac.APIWrappers.WrapperDelegateAccessor_ interface.

First, check if the wrapped object is instanceof _APIWrappers.WrapperDelegateAccessor_; 
then use the _APIWrappers.WrapperDelegateAccessor.getWrapperDelegate()_ to access the wrapped object.
I have made the _WrapperDelegateAccessor_ public for this purpose.

Reply:
Similar logic should work for accessing wrapped ProcessingEnvironment too.

Reply:
Thanks for the feedback, I tried to do it that way but it does not work. For compilation I created a stub for `WrapperDelegateAccessor`. At build time the class should be already there but lombok seems to be unable to find it and is unable to perform the cast and the method call.

Reply:
Ah, indeed, this is because JPS classes are deliberately isolated in a separate classloader from annotation processors. This is needed to avoid possible conflicts between libraries that use annotation processors and libraries used by JPS.
Then we can stick with the same idea, but use reflection to access the delegate like this:

Class<?> accessorInterface = Class.forName("org.jetbrains.jps.javac.APIWrappers$WrapperDelegateAccessor", true, proxy.getClass().getClassLoader());
Method accessor = accessorInterface.getMethod("getWrapperDelegate");
Object delegate = accessorInterface.isInstance(proxy)? accessor.invoke(proxy) : null;

The code assumes the "proxy" reference points to the proxy wrapper